giotto/mapper/visualization.py

```diff
@@ -485,30 +425,33 @@ def get_cluster_params_widgets(param, value):
         else:
             return None
 
-    def update_figure(old_figure, new_figure):
+    def update_figure(old_figure, new_figure, dim):
         # TODO could this be abstracted to node and edge traces and metadata
-        #  information without need for creating a full new figure object
+        # information without need for creating a full new figure object
         old_figure.data[0].x = new_figure.data[0].x
         old_figure.data[0].y = new_figure.data[0].y
         old_figure.data[1].x = new_figure.data[1].x
         old_figure.data[1].y = new_figure.data[1].y
+
+        if dim == 3:
+            old_figure.data[0].z = new_figure.data[0].z
+            old_figure.data[1].z = new_figure.data[1].z
+
         old_figure.data[1].marker.size = new_figure.data[1].marker.size
         old_figure.data[1].marker.color = new_figure.data[1].marker.color
         old_figure.data[1].marker.sizeref = new_figure.data[1].marker.sizeref
 
-    def get_figure(pipe, data, node_pos, node_color, columns_to_color,
-                   summary_stat):
-        graph = pipe.fit_transform(data)
-        node_elements = graph['node_metadata']['node_elements']
-        if node_pos is None:
-            node_pos = graph.layout('kamada_kawai')
+    def get_figure(pipe, data, layout, dim, color_variable,
+                   node_color_statistic, color_by_columns_dropdown,
+                   plotly_kwargs):
 
-        if node_color is None:
-            node_color = get_node_summary(node_elements, data,
-                                          summary_stat=summary_stat)
-
-        return create_network_2d(graph, data, node_pos, node_color,
-                                 columns_to_color, plotly_kwargs)
+        return create_static_network(
+            pipe, data, layout, dim,
```

Comment:
Technically, the cloning done internally by `create_static_network` is unnecessary here. Proposal: add a 'clone_pipeline' kwarg to `create_static_network`, which defaults to True, and controls whether a clone is made or not? This way we can pass False here, as cloning is already done at the beginning of the call to `create_interactive_network`.

Reply:
Good idea - I'll implement it

Reply:
I've actually opted to remove the `get_figure()` function altogether - it was previously needed to perform some intermediate calculations but with the new API it is redundant

Reply:
I agree, it is just wrapping. But I think the discussion point on cloning remains valid.